addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6729,6 +6763,35 @@ registry.ImageTools = ImageHandlerOption.extend({
      */
     async setImgShape(previewMode, widgetValue, params) {
         const img = this._getImg();
+        const restoreCursor = preserveCursor(this.$target[0].ownerDocument);
+        const document = this.el.ownerDocument;
+        const imageCropWrapperElement = document.createElement("div");
+        // During temporary mounting, making sure component isn't visible.
+        imageCropWrapperElement.classList.add("d-none");
+        document.body.append(imageCropWrapperElement);
+        const imageCropWrapper = await attachComponent(this, imageCropWrapperElement, ImageCrop, {
+            activeOnStart: true,
+            media: img,
+            mimetype: this._getImageMimetype(img),
+        });
+        await imageCropWrapper.component.mountedPromise;
+        const isCropRequired = this._isCropRequired();
+        if (previewMode === true) {
+            await imageCropWrapper.component.cropSquare(isCropRequired, previewMode);
+            img.classList.add("o_we_image_cropped_preview");
+        } else if (previewMode === false) {
+            await imageCropWrapper.component.cropSquare(isCropRequired && widgetValue);
+        } else {
+            await imageCropWrapper.component.resetCropSquare();
+            img.classList.remove("o_we_image_cropped_preview");
+        }
+        if (isGif(this._getImageMimetype(img))) {
+            img.dataset[img.dataset.shape ? "originalMimetype" : "mimetype"] = "image/png";
+        }
+        await this._reapplyCurrentShape();
+        imageCropWrapperElement.remove();
+        imageCropWrapper.destroy();
```

Comment:
Code suggestion => I made the code before having the idea of using the style for the preview  so it is still using the cropper for preview mode, but I am still posting it anyway :wink: 
So use it for:
- using the clonedImage to hide the different transformations
- the conditions limiting the need to crop the image (so only if it has a preview and if it not already a 1:1 cropped image)
- cleaner case separation 

Note that I changed the `cropSquare` and `resetCropSquare` so it already uses them (see the suggestions in the code near these functions) but again, it surely won't be needed if you use the styles instead of the cropper for previews :+1: 

```js
/**
 * @see this.selectClass for parameters
 */
async setImgShape(previewMode, widgetValue, params) {
    this.trigger_up("disable_loading_effect");
    const img = this._getImg();

    // Clone of the image used to hide the image processing (to avoid
    // flickering.)
    let clonedImgEl;

    // FIX ME:
    // - when widgetValue is undefined and preview is "reset", when
    // selecting a shape for the first time, the shape applies and then is
    // removed. It happens only when triggering `image_cropped` but this
    // event is necessary to rerender some options, so it can't be removed
    // - check the history when fixed.

    // If the shape needs the image to be square (ratio 1:1) and if not
    // already the case, crop the image before applying the shape. Reset the
    // crop when reseting the preview of such a shape.
    const isCropRequired = this._isCropRequired() && img.dataset.aspectRatio !== "1/1";
    const hasCroppedPreview = img.classList.contains("o_we_image_cropped_preview");
    if (isCropRequired || hasCroppedPreview) {
        // Mounting the cropper.
        const document = this.el.ownerDocument;
        const imageCropWrapperElement = document.createElement("div");
        // During temporary mounting, making sure component isn't visible.
        imageCropWrapperElement.classList.add("d-none");
        document.body.append(imageCropWrapperElement);
        const imageCropWrapper = await attachComponent(this, imageCropWrapperElement, ImageCrop, {
            activeOnStart: true,
            media: img,
            mimetype: this._getImageMimetype(img),
        });
        await imageCropWrapper.component.mountedPromise;

        // Replace the image by its clone.
        clonedImgEl = img.cloneNode(true);
        img.classList.add("d-none");
        img.insertAdjacentElement("afterend", clonedImgEl);

        if (previewMode !== "reset") {
            await imageCropWrapper.component.cropSquare(previewMode);
            img.classList.toggle("o_we_image_cropped_preview", previewMode);

            if (!previewMode) {
                if (isGif(this._getImageMimetype(img))) {
                    img.dataset[img.dataset.shape ? "originalMimetype" : "mimetype"] = "image/png";
                }
            }
        } else if (hasCroppedPreview) {
            await imageCropWrapper.component.resetCropSquare();
            img.classList.remove("o_we_image_cropped_preview")
        }
        imageCropWrapper.destroy();
        imageCropWrapperElement.remove();
        await this._reapplyCurrentShape();
    }

    const saveData = previewMode === false;
    if (img.dataset.hoverEffect && !widgetValue) {
        // When a shape is removed and there is a hover effect on the
        // image, we then place the "Square" shape as the default because a
        // shape is required for the hover effects to work.
        const shapeImgSquareWidget = this._requestUserValueWidgets("shape_img_square_opt")[0];
        widgetValue = shapeImgSquareWidget.getActiveValue("setImgShape");
    }
    if (widgetValue) {
        await this._loadShape(widgetValue);
        if (previewMode === 'reset' && img.dataset.shapeColors) {
            // When we reset the shape we need to reapply the colors the
            // user had selected.
            await this._applyShapeAndColors(false, img.dataset.shapeColors.split(';'));
        } else {
            // If the preview mode === false we want to save the colors
            // as the user chose their shape
            await this._applyShapeAndColors(saveData);
            if (saveData && img.dataset.mimetype !== 'image/svg+xml') {
                img.dataset.originalMimetype = img.dataset.mimetype;
                img.dataset.mimetype = 'image/svg+xml';
            }
            // When the user selects a shape, we remove the data attributes
            // that are not compatible with this shape.
            if (saveData) {
                if (!this._isTransformableShape()) {
                    delete img.dataset.shapeFlip;
                    delete img.dataset.shapeRotate;
                }
                if (!this._canHaveHoverEffect()) {
                    delete img.dataset.hoverEffect;
                    delete img.dataset.hoverEffectColor;
                    delete img.dataset.hoverEffectStrokeWidth;
                    delete img.dataset.hoverEffectIntensity;
                    img.classList.remove("o_animate_on_hover");
                }
                if (!this._isAnimatedShape()) {
                    delete img.dataset.shapeAnimationSpeed;
                }
            }
        }
    } else {
        // Re-applying the modifications and deleting the shapes
        img.src = await applyModifications(img, {mimetype: this._getImageMimetype(img)});
        delete img.dataset.shape;
        delete img.dataset.shapeColors;
        delete img.dataset.fileName;
        delete img.dataset.shapeFlip;
        delete img.dataset.shapeRotate;
        delete img.dataset.shapeAnimationSpeed;
        if (saveData) {
            img.dataset.mimetype = img.dataset.originalMimetype;
            delete img.dataset.originalMimetype;
        }
        // Also apply to carousel thumbnail if applicable.
        weUtils.forwardToThumbnail(img);
    }
    img.classList.add('o_modified_image_to_save');

    // Remove the clone and show back the image.
    if (clonedImgEl) {
        clonedImgEl.remove();
        img.classList.remove("d-none");
    }
    this.trigger_up("enable_loading_effect");
},
```